IterativeGridGenerator.cpp:43:25: error: ‘d’ not specified in enclosing ‘parallel’

sgpp fails to build with gcc 9.x:

```
optimization/src/sgpp/optimization/gridgen/IterativeGridGenerator.cpp: In member function ‘void sgpp::optimization::IterativeGridGenerator::evalFunction(size_t)’:
optimization/src/sgpp/optimization/gridgen/IterativeGridGenerator.cpp:43:25: error: ‘d’ not specified in enclosing ‘parallel’
   43 |     base::DataVector x(d);
      |                         ^
optimization/src/sgpp/optimization/gridgen/IterativeGridGenerator.cpp:41:9: error: enclosing ‘parallel’
   41 | #pragma omp parallel shared(fX, oldGridSize, gridStorage) default(none)
      |         ^~~
optimization/src/sgpp/optimization/gridgen/IterativeGridGenerator.cpp:55:9: error: ‘curGridSize’ not specified in enclosing ‘parallel’
   55 | #pragma omp for
      |         ^~~
optimization/src/sgpp/optimization/gridgen/IterativeGridGenerator.cpp:41:9: error: enclosing ‘parallel’
   41 | #pragma omp parallel shared(fX, oldGridSize, gridStorage) default(none)
      |         ^~~
scons: *** [optimization/src/sgpp/optimization/gridgen/IterativeGridGenerator.os] Error 1
optimization/src/sgpp/optimization/optimizer/unconstrained/DifferentialEvolution.cpp: In member function ‘virtual void sgpp::optimization::optimizer::DifferentialEvolution::optimize()’:
optimization/src/sgpp/optimization/optimizer/unconstrained/DifferentialEvolution.cpp:135:27: error: ‘d’ not specified in enclosing ‘parallel’
  135 |       base::DataVector y(d);
      |                           ^
optimization/src/sgpp/optimization/optimizer/unconstrained/DifferentialEvolution.cpp:132:9: error: enclosing ‘parallel’
  132 | #pragma omp parallel shared(k, a_k, b_k, c_k, j_k, prob_k, xOld, fx, fCurrentOpt, xOptIndex, \
      |         ^~~
scons: *** [optimization/src/sgpp/optimization/optimizer/unconstrained/DifferentialEvolution.os] Error 1
scons: building terminated because of errors.
```

--- a/optimization/src/sgpp/optimization/gridgen/IterativeGridGenerator.cpp
+++ b/optimization/src/sgpp/optimization/gridgen/IterativeGridGenerator.cpp
@@ -38,7 +38,7 @@ void IterativeGridGenerator::evalFunction(size_t oldGridSize) {
   const size_t curGridSize = gridStorage.getSize();
   base::DataVector& fX = functionValues;
 
-#pragma omp parallel shared(fX, oldGridSize, gridStorage) default(none)
+#pragma omp parallel shared(fX, oldGridSize, gridStorage)
   {
     base::DataVector x(d);
     ScalarFunction* curFPtr = &f;

--- a/optimization/src/sgpp/optimization/optimizer/unconstrained/DifferentialEvolution.cpp
+++ b/optimization/src/sgpp/optimization/optimizer/unconstrained/DifferentialEvolution.cpp
@@ -130,7 +130,7 @@ void DifferentialEvolution::optimize() {
     const std::vector<base::DataVector>& prob_k = prob[k];
 
 #pragma omp parallel shared(k, a_k, b_k, c_k, j_k, prob_k, xOld, fx, fCurrentOpt, xOptIndex, \
-                            xNew) default(none)
+                            xNew)
     {  // NOLINT(whitespace/braces)
       base::DataVector y(d);
       ScalarFunction* curFPtr = f.get();

--- a/optimization/src/sgpp/optimization/optimizer/unconstrained/MultiStart.cpp
+++ b/optimization/src/sgpp/optimization/optimizer/unconstrained/MultiStart.cpp
@@ -79,7 +79,7 @@ void MultiStart::optimize() {
   /*Printer::getInstance().disableStatusPrinting();
 
   #pragma omp parallel shared(d, x0, roundN, xOpt, fOpt, printer) \
-  default(none)
+ 
   {
     Optimizer* curOptimizerPtr = &optimizer;
   #ifdef _OPENMP
@@ -145,7 +145,7 @@ void MultiStart::optimize() {
     Printer::getInstance().disableStatusPrinting();
   }
 
-#pragma omp parallel shared(x0, roundN, xCurrentOpt, fCurrentOpt) default(none)
+#pragma omp parallel shared(x0, roundN, xCurrentOpt, fCurrentOpt)
   {
     UnconstrainedOptimizer* curOptimizerPtr = optimizer.get();
 #ifdef _OPENMP

--- a/optimization/src/sgpp/optimization/sle/solver/Armadillo.cpp
+++ b/optimization/src/sgpp/optimization/sle/solver/Armadillo.cpp
@@ -51,7 +51,7 @@ bool Armadillo::solve(SLE& system, base::DataMatrix& B, base::DataMatrix& X) con
   A.zeros();
 
 // parallelize only if the system is cloneable
-#pragma omp parallel if (system.isCloneable()) shared(system, A, nnz) default(none)
+#pragma omp parallel if (system.isCloneable()) shared(system, A, nnz)
   {
     SLE* system2 = &system;
 #ifdef _OPENMP

--- a/optimization/src/sgpp/optimization/sle/solver/Eigen.cpp
+++ b/optimization/src/sgpp/optimization/sle/solver/Eigen.cpp
@@ -76,7 +76,7 @@ bool Eigen::solve(SLE& system, base::DataMatrix& B, base::DataMatrix& X) const {
   size_t nnz = 0;
 
 // parallelize only if the system is cloneable
-#pragma omp parallel if (system.isCloneable()) shared(system, A, nnz) default(none)
+#pragma omp parallel if (system.isCloneable()) shared(system, A, nnz)
   {
     SLE* system2 = &system;
 #ifdef _OPENMP

--- a/optimization/src/sgpp/optimization/sle/solver/Gmmpp.cpp
+++ b/optimization/src/sgpp/optimization/sle/solver/Gmmpp.cpp
@@ -121,7 +121,7 @@ bool Gmmpp::solve(SLE& system, base::DataVector& b, base::DataVector& x) const {
     gmm::row_matrix<gmm::rsvector<double>> A(n, n);
 
 // parallelize only if the system is cloneable
-#pragma omp parallel if (system.isCloneable()) shared(system, A, nnz) default(none)
+#pragma omp parallel if (system.isCloneable()) shared(system, A, nnz)
     {
       SLE* system2 = &system;
 #ifdef _OPENMP

--- a/optimization/src/sgpp/optimization/sle/solver/UMFPACK.cpp
+++ b/optimization/src/sgpp/optimization/sle/solver/UMFPACK.cpp
@@ -80,7 +80,7 @@ bool UMFPACK::solve(SLE& system, base::DataMatrix& B, base::DataMatrix& X) const
   std::vector<double> Tx;
 
 // parallelize only if the system is cloneable
-#pragma omp parallel if (system.isCloneable()) shared(system, Ti, Tj, Tx, nnz) default(none)
+#pragma omp parallel if (system.isCloneable()) shared(system, Ti, Tj, Tx, nnz)
   {
     SLE* system2 = &system;
 #ifdef _OPENMP

